src/route/userActivity.route.ts

```diff
@@ -62,6 +63,27 @@ const userActivityRouter = () => {
     }
   );
 
+  router.patch(
+    '/activities/:activityId/cancel',
+    authorizeMiddleware,
+    async (req: Request, res: Response) => {
+      /* #swagger.tags = ['Activity'] */
+
+      const userId = req.user?._id;
+      const activityId = new mongoose.Types.ObjectId(req.params.activityId);
+      try {
+        const data = await UserActivityService.cancel(activityId);
+
+        if (!userId?.equals(data?.creatorId))
+          throwAPIError({ res, error: 'Unauthorized', statusCode: 401 });
```

Comment:
Although here the error is thrown, the status of activity is already update

Reply:
updated